src/webots/nodes/utils/WbSupervisorUtilities.cpp

```diff
@@ -1133,6 +1149,37 @@ void WbSupervisorUtilities::handleMessage(QDataStream &stream) {
       }
       return;
     }
+    case C_SUPERVISOR_FIELD_GET_FROM_INDEX: {
+      int node_id, field_index;
+      unsigned char allowSearchInProto;
+      stream >> node_id;
+      stream >> field_index;
+      stream >> allowSearchInProto;
+
+      mFoundFieldId = -1;
+      mFoundFieldType = 0;
+      mFoundFieldCount = -1;
+      mFoundFieldIsInternal = false;
+
+      WbNode *const node = WbNode::findNode(node_id);
+      if (node) {
+        WbField *field = node->field(field_index, allowSearchInProto == 1);
+        if (field) {
+          WbMultipleValue *mv = dynamic_cast<WbMultipleValue *>(field->value());
+          mFoundFieldCount = mv ? mv->size() : -1;
+          mFoundFieldId = field_index;
```

Comment:
Should we call it ID or index?

Reply:
Index is the correct term for the public API: it represents the item at a given index in the node's list of field.
Fields objects don't have ids.
However the already existing code is (wrongly) referring to id also for fields (in `WbSupervisorUtilities.cpp` and `supervisor.c`).

In this PR I decided to not update/fix all the existing code variable and function names because it will cause many unrelated changes. But I can do it if you prefer.